catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/EmailSenderImpl.java

```diff
@@ -0,0 +1,127 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.alliance.nsili.endpoint;
+
+import static org.apache.commons.lang3.Validate.notNull;
+
+import java.util.Properties;
+
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EmailSenderImpl implements EmailSender {
+
+    private static final String SMTP_HOST_PROPERTY = "mail.smtp.host";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EmailSenderImpl.class);
+
+    private String fromEmail;
+
+    private String mailHost;
+
+    private String subject;
+
+    private String emailBody;
+
+    /**
+     * EmailSenderImpl formats emails and sends them to the desired address using the fromEmail
+     * and through the desired mailHost.
+     *
+     * @param fromEmail: non-null origin address
+     * @param mailHost:  non-null origin SMTP mailHost to send through
+     * @param subject:   non-null subject line of email to be sent
+     * @param emailBody: non-null body of email to be sent
+     */
+    public EmailSenderImpl(String fromEmail, String mailHost, String subject, String emailBody) {
+        notNull(fromEmail, "fromEmail must be non-null");
+        notNull(mailHost, "mailHost must be non-null");
+        notNull(subject, "subject must be non-null");
+        notNull(emailBody, "emailBody must be non-null");
+
+        this.fromEmail = fromEmail;
+        this.mailHost = mailHost;
+        this.subject = subject;
+        this.emailBody = emailBody;
+    }
+
+    @Override
+    public void sendEmail(String emailDest) {
+        try {
+            InternetAddress emailAddr = new InternetAddress(emailDest);
+            emailAddr.validate();
+
+            Properties properties = createSessionProperies();
+
+            Session session = Session.getDefaultInstance(properties);
+            MimeMessage mimeMessage = new MimeMessage(session);
+            mimeMessage.setFrom(new InternetAddress(fromEmail));
+            mimeMessage.addRecipient(Message.RecipientType.TO, emailAddr);
+            mimeMessage.setSubject(subject);
+            mimeMessage.setText(emailBody);
+            Transport.send(mimeMessage);
+
+        } catch (AddressException e) {
+            LOGGER.debug("Invalid email address entered", e);
+        } catch (MessagingException e) {
+            LOGGER.debug("Message error occured on send", e);
+        }
+    }
+
+    private Properties createSessionProperies() {
+        Properties properties = System.getProperties();
+        properties.setProperty(SMTP_HOST_PROPERTY, mailHost);
+        return properties;
+    }
+
+    @Override
+    public String toString() {
+        notNull(fromEmail, "fromEmail must be non-null");
```

Comment:
I don't know if I agree with having notNull here.  If `fromEmail` is null, we can't see the toString?